### PR TITLE
Vsync: protect tasks in the same pool from failing each other

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: src/service/vsync-impl.js [hunk 1]
```diff
@@ -375,12 +375,22 @@ export class Vsync {
     this.states_ = this.nextStates_;
     for (let i = 0; i < tasks.length; i++) {
       if (tasks[i].measure) {
-        tasks[i].measure(states[i]);
+        try {
```
Comment (cramforce) [inline src/service/vsync-impl.js]: Consider making a `*NoInline` function to avoid the try block optimizing the function. Not sure it matters here.

Comment (dvoytenko) [inline src/service/vsync-impl.js]: Done. PTAL.

File: src/service/vsync-impl.js [hunk 2]
```diff
@@ -416,6 +419,23 @@ export class Vsync {
 
 
 /**
+ * For optimization reasons to stop try/catch from blocking optimization.
+ * @param {function(!VsyncStateDef)|undefined} callback
```
Comment (jridgewell) [inline src/service/vsync-impl.js]: I don't think it can be `undefined`, we check specifically for `task.measure` (and `mutate`) before calling.

Comment (dvoytenko) [inline src/service/vsync-impl.js]: Unfortunately, this is the case. @erwinmombay is working on this.

**Pull request merged**